src/test/java/org/codehaus/plexus/util/xml/pull/MXParserTest.java

```diff
@@ -156,6 +158,46 @@ public void testUnicodeEntities()
         assertEquals( XmlPullParser.END_TAG, parser.nextToken() );
     }
 
+    @Test
+    public void testInvalidCharacterReferenceHexa()
+        throws Exception
+    {
+        MXParser parser = new MXParser();
+        String input = "<root>&#x110000;</root>";
+        parser.setInput( new StringReader( input ) );
+
+        try
+        {
+            assertEquals( XmlPullParser.START_TAG, parser.nextToken() );
+            assertEquals( XmlPullParser.ENTITY_REF, parser.nextToken() );
+            fail( "Should fail since &#x110000; is an illegal character reference" );
+        }
+        catch ( XmlPullParserException e )
+        {
+            assertTrue( e.getMessage().contains( "character reference (with hex value 110000) is invalid" ) );
+        }
+    }
+
+    @Test
+    public void testInvalidCharacterReferenceDecimal()
```

Comment:
when reading the code change (but I didn't try to execute and check), it seems valid decimal character references were simply ignored
permahp adding a unit test for valid character references would be useful also :)

Reply:
Thanks! This actually uncovered a bug when parsing decimal char refs >=65536. Fixed it too.

Reply:
great news!
thanks for the quick fix: I'll be able to release today!!!